### ISSUE TITLE
Crash on compile

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1142]
Version 0.43.1 (Nightly 08/27/2021)
Unable to isolate the issue, but here is [crash dump](https://github.com/beefytech/Beef/files/7072820/crash_on_compile.zip).

**Comment by Fusioon** [comment]
Alright, after some backtracking i was able to identify what caused this.
Code to reproduce:
```
enum Filter
{
    Linear,
    Nearest
}
abstract class Texture
{
    public abstract (Filter min, Filter mag) Filter { get; set; }
}

class Texture2D : Texture
{
    Filter _min, _mag;
}

extension Texture2D
{
    public override (Filter min, Filter mag) Filter 
    {
        get => (_min, _mag);
        set => (_min, _mag) = value;
    }

}

class Program
{
    static void Main(String[] args)
    {
        Texture2D texture = scope .();
        texture.Filter = (.Linear, .Linear);
    }
}	
```

**Comment by bfiete** [comment]
Fixed at 25a71c284a516e532f7ff72e7429c088c75aea74
